### PR TITLE
ci: run all check CIs against patch PR

### LINKED ISSUE #640: build action should also be ran against PRs targeting release branch
Otherwise we need to manually run build task before merging any patch PR

### PR DESCRIPTION
**Pull request by qweeah**
I have validated it in my own repo: [build](https://github.com/qweeah/oras/actions/runs/3225144165)

Resolves #640 

Signed-off-by: Billy Zha <jinzha1@microsoft.com>

### CODE REVIEW

**Review by shizhMSFT — CHANGES_REQUESTED** [review]
How about others? like CodeQL and license checker?

**Comment by qweeah** [comment]
> How about others? like CodeQL and license checker?

Good point, updated.

File: .github/workflows/license-checker.yml [hunk 1]
```diff
@@ -17,7 +17,9 @@ on:
   push:
     branches: main
```
Comment (shizhMSFT) [inline .github/workflows/license-checker.yml:18]: How about `push`?

Comment (qweeah) [reply .github/workflows/license-checker.yml]: You are right! Added.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**